samples/java/src/main/java/com/example/JsonIdAdapter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.example;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.PrivateKey;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import java.util.logging.Logger;
+import javax.json.Json;
+import javax.json.JsonObject;
+
+import org.hyperledger.fabric.client.identity.Identities;
+import org.hyperledger.fabric.client.identity.Identity;
+import org.hyperledger.fabric.client.identity.Signer;
+import org.hyperledger.fabric.client.identity.Signers;
+import org.hyperledger.fabric.client.identity.X509Identity;
+
+/**
+ * This class can be used to map identities in a variety of JSON formats to the
+ * Identity and Signers required for the gateway. For example if you have an
+ * application wallet, or have exported IDs from SaaS
+ * 
+ * ``` 
+ * // setup the loading of the identities JsonIdAdapter idWallet = new
+ * JsonIdAdapter(Paths.get(walletDir)); Identity id =
+ * idWallet.getIdentity(userid); Signer signer = idWallet.getSigner(userid);
+ * Gateway.Builder builder =
+ * Gateway.newInstance().identity(id).signer(signer).connection(grpcChannel);
+ * 
+ * ```
```

Comment:
JavaDoc doesn't do markdown do you probably want:
```
<pre>{@code
// Code example here
}</pre>
```
And perhaps format so lines don't mix comments with trailing code, or contain multiple commands.

Reply:
For the example it may be more natural to create a Gateway instance rather than the Builder, which would match up with what you have for the Node sample:
```
Gateway gateway = Gateway.newInstance()
        .identity(id)
        .signer(signer)
        .connection(grpcChannel)
        .connect();
```